topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,415 @@
-[//]: # (title: Integrate KMM into an existing application)
-[//]: # (auxiliary-id: Integrate_KMM_into_an_existing_application)
+[//]: # (title: Make your Android application work on iOS – tutorial)
+[//]: # (auxiliary-id: Make_your_Android_application_work_on_iOS_tutorial)
 
-Here you can learn how to integrate Kotlin Multiplatform Mobile (KMM) into an existing Android application and make it multiplatform so that it works on both Android and iOS. 
-You will understand what kind of code is good to share and how to use shared code in your existing application.
+Here you can learn how to make your existing Android application cross-platform so that it works both on Android and iOS.
+You'll be able to write code and test it for both Android and iOS only once, in one place.
 
-Sharing code saves you time and effort on writing code and testing it for Android and iOS as you only need to do it once, in the one place.
+This tutorial uses a [sample Android application](https://github.com/Kotlin/kmm-integration-sample) with a single screen for entering a username and password.
+The credentials are validated and saved to an in-memory database.
 
-Before you begin, learn how to [create and configure a KMM application](create-first-app.md).
+If you aren't familiar with KMM, you can learn how to [create and configure a KMM application from scratch](create-first-app.md) first.
 
-## Decide what to share
+## Prepare an environment for development
 
-An application includes two layers:
+1. Install Android Studio 4.2 or higher and [other tools for KMM development](setup.md) on macOS.
 
-* The **frontend** that defines the user interface and its behavior, including animations and transitions.
-* The **backend** that defines the business logic, data management, network stack, and other components that support the frontend.
+   >You will need a Mac with macOS to complete certain steps in this tutorial, which include writing iOS-specific code and running an iOS application.  
+   >These steps cannot be performed on other operating systems, such as Microsoft Windows. This is due to an Apple requirement.
+   >
+   {type="note"}
 
-KMM presumes that the best user experience is when it is native to the platform itself, and therefore we don’t recommend sharing it in KMM. 
-However, you can share the code for the UI behavior that defines what happens with any user interaction and how the frontend communicates 
-with the backend with the help of compatible architectural patterns.
+2. In Android Studio, create a new project from version control: `https://github.com/Kotlin/kmm-integration-sample`.
```

Comment:
minor: This sentence looks strange on the staging site because of the line break.

Reply:
There is no a line break in the source. This is because the code is long and it is non-breakable. Nothing to do here 🤷‍♀️